### DISCUSSION TITLE
Angular ou React em 2020?

### DISCUSSION THREAD

**Discussion by kazzkiq** [frontendbr/forum#1713] [Discussão]
Gostaria de propor uma discussão interessante, e *levemente* polêmica sobre qual tecnologia escolher para começar um projeto grande em 2020.

Imagine o seguinte cenário:

**Você e sua equipe possuem conhecimento em React e Angular. Sua empresa fecha um projeto com um cliente que está aberto ao uso de qualquer uma das duas tecnologias. Hoje, começando um projeto do zero, você escolheria Angular ou React?**

**Comment by danilosilvadev** [comment]
Possuo conhecimentos em angular, react e vuejs. Sendo que react foi a stack que passei mais tempo e fiz mais projetos. Se vc quer algo rápido e fácil vuejs é PERFEITO para essa task, mas se vc quer enforce design patterns, escalabilidade, tipagem estática e todas aquelas firulas que fazem um projeto GRANDE ser passível de quickfixes sem ler extensa documentação, a resposta é Angular.

Para mim react é bom em fazer marketing e dat is it. Não é simples e nem faz sentido como vuejs e n é organizado nem completo como angular, sinceramente n entendo pq o mercado escolhe tanto react.

**Comment by ninetails** [comment]
> Hoje, começando um projeto do zero, você escolheria Angular ou React?

Nessa pergunta pessoal, eu escolheria **React**

> Você e sua equipe possuem conhecimento em React e Angular.

Mas dado o contexto geral, ambas as opções são factíveis e como a decisão deveria ser dos **devs**, e todos sabem ambos, faz todo mundo se degladiarem nessa guerra de _fanboyzisse_ usando o projeto do novo cliente como parâmetro pra galera desse novo time discutir qual ficaria melhor e usa o que ganhar. E quem não participar, não tem opinião. E fora que serão eles quem darão manutenção, logo que todos num consenso escolham qual teria menos atrito dentro da equipe.

**Comment by andrefelizardo** [comment]
Eu ainda ficaria preso ao contexto do projeto. O projeto vai precisar de muitas customizações de build? A arquitetura precisa ser muito disruptiva? 

Se fosse pessoal mesmo seria: qual framework você usaria para criar um projeto só seu?

**Comment by maucrvlh** [comment]
Considerando mercado, possibilidade de pegar freelas p frente ou mesmo
outras oportunidades, eu iria de React (infelizmente o mercado adotou).

Se fosse por questões pessoais, eu não iria de Angular (apesar de bem
maduro, tenho a impressão de não estar tão na crista da onda quanto os
outros dois maiores). Pessoalmente eu iria de Vue.js. Tão poderoso quanto
React, leve, fácil, comunidade imensa e uma toolbox recheada de facilidades
ao dev.

**Comment by LuanDc** [comment]
O Angular é um Framework excepcional. Mas ele claramente foi feito pra ser adotado em grandes projetos. Outro problema é que por ter tanto Pattern, o Angular só vai fazer sentido pra quem realmente entende o porquê aquilo é necessário. É uma pergunta difícil de ser respondida, depende da tua equipe, do teu escopo de projeto, escalabilidade e etc. Particularmente, eu prefiro o React por me sentir mais a vontade com programação funcional, mas é só opinião pessoal do que necessariamente uma verdade absoluta.

**Comment by lucasferreiralimax** [comment]
Primeiro precisaria saber o escopo do projeto e sua complexidade.

O tamanho do projeto e suas features é crucial para uma decisão mais sensata e a sustentação do mesmo.

Mas de fato é uma decisão comercial difícil precisar analisar o time e entender a capacidade técnica e agilidade de cada desenvolvedor.

## Angular
Um framework completo que vem crescendo cada vez mais com novidades, organização do projeto e seus padrões.
Tem um CLI que ajuda no desenvolvimento e implementações
Baseado em componentes e módulos
Não precisa implementar muitas lib
Comunidade grande para obter ajudar

## React
Uma lib de interface que abstrai bastante a capacidade do JavaScript dando muitas possibilidades e muitos patterns
Também vem crescendo muito
Baseado em componentes
Precisa complementar com outras lib conforme aumenta a complexidade
Comunidade grande para obter ajudar

## Vue
#### Meu velho amigo
Um framework completo que está na frente de muitos conceitos com uma sintaxe fácil de aplicar e aprender
Gosto muito desse por ser bem agnóstico
Baseado em componentes
Comunidade grande para obter ajudar

## Webpack, Parcel.js, Gulp, Grunt
Criar suas próprias rotinas automatizadas, minificações, template engine, pos e pre-processadores.

## HTML, CSS, JavaScript
Se for algo simples ou muito especifico.

___
Bom essa é minha humilde opinião, existe muitos projetos legais para se trabalhar e desenvolver soluções de pequena ou grande escala é necessário sempre questionar os pros e contras sobre atualizações como cada projeto pode impactar sua solução futuramente em manutenção.

Espero não ter fugido muito do assunto
Bons estudos para todos :-)
